src/js/core/directives/ui-grid.js

```diff
@@ -68,13 +71,10 @@
       function columnDefsWatchFunction(n, o) {
         if (n && n !== o) {
           self.grid.options.columnDefs = $scope.uiGrid.columnDefs;
-          self.grid.buildColumns({ orderByColumnDefs: true })
-            .then(function(){
-
-              self.grid.preCompileCellTemplates();
-
-              self.grid.callDataChangeCallbacks(uiGridConstants.dataChange.COLUMN);
-            }).catch(angular.noop);
+          self.grid.callDataChangeCallbacks(uiGridConstants.dataChange.COLUMN, {
```

Comment:
Prevent buildColumns from being triggered multiple time during initialization. See lines 598 and 601 of [Grid.js](https://github.com/mportuga/ui-grid/blob/75f98c3eda3e46f14f9d924e6f5a6d6e35bcc2b5/src/js/core/factories/Grid.js#L601).